windows/RNZipArchiveModule.cpp

```diff
@@ -0,0 +1,175 @@
+//  RNZipArchive
+
+#include "pch.h"
+
+#include "RNZipArchiveModule.h"
+
+#include "miniz.h"
+#include "miniz_zip.h"
+
+#include <filesystem>
+#include <winrt/Windows.Storage.h>
+
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Foundation::Collections;
+using namespace winrt::Windows::System;
+using namespace winrt::Windows::System::Threading;
+
+void RNZipArchiveModule::isPasswordProtected(
+    std::string zipFilePath, winrt::Microsoft::ReactNative::ReactPromise<bool> promise) noexcept
+{
+    promise.Reject("Not Yet Implemented: isPasswordProtected");
+}
+
+winrt::fire_and_forget RNZipArchiveModule::unzip(
+    std::string zipFilePath,
+    std::string destinationDirectory,
+    std::string,
+    winrt::Microsoft::ReactNative::ReactPromise<std::string> promise) noexcept
+try
+{
+    co_await ThreadPool::RunAsync(WorkItemHandler([zipFilePath, destinationDirectory](IAsyncAction operation) {
+        mz_zip_archive archive = {};
+
+        if (operation.Status() != AsyncStatus::Started)
+        {
+            // already cancelled or otherwise doesn't need to run so bail out before doing a bunch of work
+            return;
+        }
+
+        try
+        {
+            if (MZ_TRUE != mz_zip_reader_init_file(&archive, zipFilePath.c_str(), 0))
+            {
+                throw std::invalid_argument(mz_zip_get_error_string(mz_zip_get_last_error(&archive)));
+            }
+
+            auto destinationPath = std::filesystem::path(destinationDirectory);
+            destinationPath.make_preferred();
+
+            // For whatever reason, UWPs can't make use of current_path so instead
```

Comment:
Note: The reason for this is because UWPs needs explicit popup to user to select an arbitrary folder/file
outside of app's data folder. A file/folder windows api created popup returns a StorageFile object that must be used to be able to read/write to this location. This can be difficult to properly achieve inside a RNW app because of the
separation of RN modules from one another. 

As of RNW 0.62 the needed ReactContext's Dictionary Bag to pass around such objects(StorageFile) as not yet been implemented. 

No user interaction is needed to read/write from app's data folder until then.

Reply:
I had proposed a similar but different problem here.

https://github.com/microsoft/react-native-windows/issues/4396#issuecomment-603817368
In this particular case, I simply needed to pass an object from the App.cpp into the cpp of a RN module.

While that was a different problem, the passing of arbitrary objects around via a ReactContext is the similar part.  The difference is needing some kind of PopupFile Selector Manager or something. 

Since react-native-zip-archive has no front end/UI,  something else would need to likely capture the StorageFile in a pop up somewhere and then stash the returned C++/Winrt StorageFile in some location for it to be passed in to this RN module.  Its a hard problem, and having a generalized solution is not going to be easy, until RNW adds better support for it.